src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -401,6 +401,79 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
+            assert(op2->OperIsHWIntrinsic());
+
+            // Get the registers and intrinsics that needs embedded mask
+            const HWIntrinsic intrinOp2(op2->AsHWIntrinsic());
+            instruction       insOp2     = HWIntrinsicInfo::lookupIns(intrinOp2.id, intrinOp2.baseType);
+            const bool        instrIsRMW = op2->isRMWHWIntrinsic(compiler);
+
+            regNumber maskReg     = op1Reg;
+            regNumber instrOp1Reg = REG_NA;
+            regNumber instrOp2Reg = REG_NA;
+            regNumber falseReg    = op3Reg;
+
+            switch (intrinOp2.numOperands)
+            {
+                case 2:
+                    assert(intrinOp2.op2 != nullptr);
+                    instrOp2Reg = intrinOp2.op2->GetRegNum();
+                    FALLTHROUGH;
+
+                case 1:
+                    assert(intrinOp2.op1 != nullptr);
+                    instrOp1Reg = intrinOp2.op1->GetRegNum();
+                    break;
+
+                default:
+                    unreached();
+            }
+
+            switch (intrinOp2.numOperands)
+            {
+                case 1:
+                    GetEmitter()->emitIns_R_R_R(insOp2, emitSize, targetReg, maskReg, instrOp1Reg, opt);
+                    break;
+
+                case 2:
+
+                    assert(instrIsRMW);
+                    if (targetReg != falseReg)
+                    {
+                        assert(targetReg != instrOp1Reg);
+                        assert(targetReg != instrOp2Reg);
+
+                        if (!intrin.op3->IsVectorZero())
+                        {
+                            // If the `falseValue` of conditional-select is non-zero, then we start with that in the
+                            // destination followed by `movprfx` (see below) of the relevant bits of the first operand
+                            // of `insOp2` based upon the value of predicate `cond`.
+                            GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, falseReg,
+                                                      /* canSkip */ true);
+                        }
+                    }
+
+                    // Move the first operand of `insOp2` in the destination. If `falseValue` was zero, then we can just
+                    // use /Z during the move, otherwise merge using /M into the destination.
+                    GetEmitter()->emitIns_R_R_R(INS_sve_movprfx, emitSize, targetReg, maskReg, instrOp1Reg, opt,
+                                                intrin.op3->IsVectorZero() ? INS_SCALABLE_OPTS_NONE
+                                                                           : INS_SCALABLE_OPTS_PREDICATE_MERGE);
+
+                    // Finally, perform the actual operation so that `targetReg` is the first operand and `instrOp2Reg`
+                    // having the second operand.
+                    GetEmitter()->emitIns_R_R_R(insOp2, emitSize, targetReg, maskReg, instrOp2Reg, opt);
```

Comment:
`opt` should never include `PREDICATE_MERGE` here, correct? If it did then we would have been able to elide a `movprfx`, so I think we can just assert that it doesn't.

Reply:
We use `insScalableOpts` to pass that information and in this case, we pass it as `NONE` and not `INS_SCALABLE_OPTS_PREDICATE_MERGE`.